File: 프로그래머스/unrated/120907. OX퀴즈/OX퀴즈.py
def solution(quiz):
    result = []  # 결과를 저장할 배열 초기화
    for equation in quiz:
        # 수식을 공백을 기준으로 분해하여 피연산자와 연산자를 추출
        parts = equation.split()
        X, operator, Y, equals_sign, Z = parts  # 변수에 각 부분 할당
        
        # 연산 수행
        if operator == '+':
            calc_result = int(X) + int(Y)
        else:
            calc_result = int(X) - int(Y)
        
        # 계산 결과와 주어진 결과를 비교
        if calc_result == int(Z):
            result.append('O')
        else:
            result.append('X')
    return result

File: 프로그래머스/0/120907. OX퀴즈/OX퀴즈.java
class Solution {
    public String[] solution(String[] quiz) {
        String[] answer = new String[quiz.length];
        String[] s = new String[quiz.length];
        for(int i=0;i<quiz.length;i++){
            s=quiz[i].split(" ");
            if(s[1].equals("-")){
                if(Integer.parseInt(s[0])-Integer.parseInt(s[2])== Integer.parseInt(s[4])){
                    answer[i]="O";
                }else {
                    answer[i]="X";
                }
            }
            if(s[1].equals("+")){
                if(Integer.parseInt(s[0])+Integer.parseInt(s[2])== Integer.parseInt(s[4])){
                    answer[i]="O";
                }else{
                    answer[i]="X";
                }
            }
        }
        return answer;
    }
}